.travis.yml

```diff
@@ -14,12 +14,26 @@ addons:
 cache:
   directories:
     - $HOME/.cabal
-ghc:
-  - '8.10'
-  - '8.8'
-  - '8.6'
+    - $HOME/.stack
+    - $TRAVIS_BUILD_DIR/.stack-work
+
+matrix:
+  include:
+
+  # Cabal
+  - ghc: '8.10'
+  - ghc: '8.8'
+  - ghc: '8.6'
+
+  # Stack
+  - ghc: '8.10.4'
+    env: STACK_YAML="stack.yaml"
```

Comment:
Sorry, missed this comment. This will work. Some disadvantages are that you don't get separate report on whether cabal or stack failed, but this can be seen in the logs, and for stack users one won't be able to do just `stack build` but will have to specify the stack file to use, which is also not a big deal.

Reply:
> Some disadvantages are that you don't get separate report on whether cabal or stack failed, but this can be seen in the logs

True, that's a disadvantage. Less Travis jobs though.

> and for stack users one won't be able to do just stack build but will have to specify the stack file to use

I thought about that and added a symlink `stack.yaml -> stack.8.10.3.yaml`, that should work, right?

Reply:
I tried it and it works ! :)